github/set_output.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2022 SLSA Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package github
+
+import (
+	"errors"
+	"os"
+)
+
+// SetOutput writes a name value pair to a file located at GITHUB_OUTPUT.
+func SetOutput(name, value string) error {
+	if filename := os.Getenv("GITHUB_OUTPUT"); filename != "" {
+		f, err := os.OpenFile(filename, os.O_APPEND|os.O_WRONLY, 0o755)
+		if err != nil {
+			return err
+		}
+		if _, err := f.WriteString(name + "=" + value + "\n"); err != nil {
+			return err
+		}
+		return f.Close()
+	}
+	return errors.New("file GITHUB_OUTPUT not found")
```

Comment:
nit: I think general consensus for readability nowadays is to exit early for errors like this rather than at the end.
```suggestion
	filename := os.Getenv("GITHUB_OUTPUT")
	if filename == "" {
		return errors.New("file GITHUB_OUTPUT not found")
	}
	f, err := os.OpenFile(filename, os.O_APPEND|os.O_WRONLY, 0o755)
	if err != nil {
		return err
	}
	if _, err := f.WriteString(name + "=" + value + "\n"); err != nil {
		return err
	}
	return f.Close()
```

Reply:
nit: Alternatively you could fall back to using `set-output` if `GITHUB_OUTPUT` is not defined as they do in the Github Actions toolkit
https://github.com/actions/toolkit/blob/b36e70495fbee083eb20f600eafa9091d832577d/packages/core/src/core.ts#L192-L200

Reply:
I'll add a TODO to remove this when it's EOL for set-output